disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,252 @@
+import os
+import re
+import yaml
+
+try:
+    import pooch
+    import tqdm
+except ImportError:
+    raise ImportError('Please install disdrodb with "dev" dependencies to use this module.')
+
+from disdrodb.api import io
+
+
+# function to check if a path exists
+def check_path(path: str) -> None:
+    """Check if a path exists.
+
+    Parameters
+    ----------
+    path : str
+        Path to check.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the path does not exist.
+    """
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"The path {path} does not exist. Please check the path.")
+
+
+def get_metadata_dirs(dir_path: str) -> list:
+    """Return a list containing all metadata folders paths.
+
+
+    Parameters
+    ----------
+    dir_path : str
+        Folder path.
+
+    Returns
+    -------
+    list
+        List of metadata folders.
+
+    """
+
+    return [os.path.join(root, "metadata") for root, dirs, files in os.walk(dir_path) if "metadata" in dirs]
+
+
+def get_yaml_files(metadata_dirs: str) -> list:
+    """Return a list containing all YAML files paths.
+
+
+    Parameters
+    ----------
+    metadata_dirs : str
+        Folder path.
+
+    Returns
+    -------
+    list
+        List of yaml files.
+    """
+
+    return [
+        os.path.join(root, file)
+        for metadata_dir in metadata_dirs
+        for root, dirs, files in os.walk(metadata_dir)
+        for file in files
+        if file.endswith(".yml") or file.endswith(".yaml")
+    ]
+
+
+def get_sation_name_and_url_from_metadata(yaml_file_path: str) -> dict:
+    """Return the station name and the url from the metadata folder.
+
+    Parameters
+    ----------
+    yaml_file_path : str
+        Path to the metadata yaml file.
+
+    Returns
+    -------
+    dict
+        Dictionnary containing the station name and the url.
+    """
+
+    result_dict = {}
+    with open(yaml_file_path) as f:
+        metadata_dict = yaml.safe_load(f)
+

```

Comment:
Here check that the `station_name` key is equal to `os.path.basename(yaml_file_path).replace(".yml","")`